### PR TITLE
deploy: prefer caliper v2.9.1 due to segfault issue with the 2.10.0

### PR DESCRIPTION
**Pull request by pramodk**
I will report the issue upstream. For now, lets use the older version that was working OK.

Edit: LLNL/Caliper/issues/529

### CODE REVIEW

File: bluebrain/deployment/environments/applications_libraries.yaml [hunk 1]
```diff
@@ -36,7 +36,7 @@ spack:
           all: '{name}/{version}'
   specs:
     - boost+atomic+chrono+date_time+filesystem+json+locale+log+math+program_options+python+random+regex+serialization+shared+signals+stacktrace+system+test+timer+type_erasure
-    - caliper+cuda cuda_arch=70
+    - caliper+cuda@2.8.0 cuda_arch=70
```
Comment (pramodk) [inline bluebrain/deployment/environments/applications_libraries.yaml]: @matz-e : This is not sufficient to enforce 2.8.0 as preferred. What is the current way to set the preferred version? 
* I guess we no longer have packages.yaml
* Should I update the recipe in the builtin repo directly? Or, just create one in repo-patches?

**Comment by pramodk** [comment]
Multiscale-Run module check is failing but that's irrelevant for this PR.

**Review by matz-e — CHANGES_REQUESTED** [review]
No. This goes under the `packages:caliper:require:@2.91` setting in `bluebrain/sysconfig/bluebrain5/packages.yaml`.

File: bluebrain/sysconfig/bluebrain5/packages.yaml [hunk 2]
```diff
@@ -8,6 +8,9 @@ packages:
     variants: +gas+gold+ld+plugins~libiberty
   boost:
     variants: +filesystem+pic+serialization+test
+  caliper:
+    # see https://github.com/LLNL/Caliper/issues/529
+    version: [2.9.1]
```
Comment (matz-e) [inline bluebrain/sysconfig/bluebrain5/packages.yaml]:
```suggestion
    require: "@2.9.1"
```
for a "hard" requirement

**Review by matz-e — DISMISSED** [review]

**Review by matz-e — APPROVED** [review]

**Pull request merged**